.github/workflows/sonarqube.yml

```
@@ -12,16 +12,71 @@ jobs:
+      ALLOWED_HOSTS: "localhost,127.0.0.1"
+      # Set SONAR_TOKEN for the SonarCloud step
+      SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+
     steps:
       - name: Checkout code
         uses: actions/checkout@v3
         with:
           fetch-depth: 0  # Important for SonarQube to see history
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
```

actionlint (1.7.4)

the runner of "actions/setup-python@v4" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)